packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_view.dart

```diff
@@ -13,6 +18,173 @@ class SnapshotView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Text('Content of ${item.name} will be here');
+    return ValueListenableBuilder<bool>(
+      valueListenable: item.isProcessing,
+      builder: (_, isProcessing, __) {
+        if (isProcessing) return const SizedBox.shrink();
+
+        final stats = item.stats;
+        if (stats == null) {
+          return const Center(child: Text('Could not take snapshot.'));
+        }
+
+        return _StatsTable(
+          data: stats,
+          sorting: item.sorting,
+          key: ObjectKey(item),
```

Comment:
nit: order key first. Can you also add a comment explaining why we need a key here?